src/MOIWrapper.jl

```diff
@@ -1,408 +1,435 @@
-    objective_coefficients = zeros(Float64, get_num_cols(instance.inner))
-    for (col, coef) in zip(cols, coefs)
-        objective_coefficients[col] += coef
+    # Create linear constraints
+    nrows = 0  # number of rows in the problem
+    for (F, S) in MOI.get(src, MOI.ListOfConstraints())
+        # Check if constraint is a linear constraint
+        (F == MOI.ScalarAffineFunction{Float64}) || continue
+
+        for index in MOI.get(src, MOI.ListOfConstraintIndices{F, S}())
+            # Retrieve problem data
+            # TODO: ensure that `(f, s)` are in canonical form
+            f = MOI.get(src, MOI.ConstraintFunction(), index)
+            s = MOI.get(src, MOI.ConstraintSet(), index)
+
+            # Extract bounds
+            # Constraint writes a'x + a0 - in - set
+            # So we offset the bounds by -a0
+            lb, ub = _bounds(s) .- f.constant
+
+            # Identify constraint terms
+            coeffs = [t.coefficient for t in f.terms]
+            cols = [
+                mapping.varmap[t.variable_index].value
+                for t in f.terms
+            ]
+
+            # Add constraint
+            Clp.add_row(dest.inner, Cint(length(f.terms)), Cint.(cols .- 1), coeffs, lb, ub)
+
+            # Track index
+            mapping.conmap[index] = MOI.ConstraintIndex{F, S}(nrows + 1)
+            nrows += 1
+        end
     end
-    chg_obj_coefficients(instance.inner, objective_coefficients)
-end
 
-function LQOI.change_objective_sense!(instance::Optimizer, sense::Symbol)
-    if sense == :min
-        set_obj_sense(instance.inner, 1.0)
-    elseif sense == :max
-        set_obj_sense(instance.inner, -1.0)
+    # Objective function
+    sense = MOI.get(src, MOI.ObjectiveSense())
+    if sense == MOI.MIN_SENSE
+        Clp.set_obj_sense(dest.inner, 1)   # minimize
+    elseif sense == MOI.MAX_SENSE
+        Clp.set_obj_sense(dest.inner, -1)  # maximize
+    elseif sense == MOI.FEASIBILITY_SENSE
+        Clp.set_obj_sense(dest.inner, 0)   # feasibility
     else
-        error("sense must be either :min or :max")
+        error("Unknown objective sense: $(sense).")
     end
-end
-
-function LQOI.get_linear_objective!(instance::Optimizer, x::Vector{Float64})
-    copyto!(x, get_obj_coefficients(instance.inner))
-end
 
-function LQOI.solve_linear_problem!(instance::Optimizer)
-    solveroptions = ClpSolve()
-    model = instance.inner
-    for (name, value) in instance.params
-        if haskey(optionmap, name)
-            optionmap[name](model, value)
-        elseif haskey(solveoptionmap, name)
-            solveoptionmap[name](solveroptions,value)
-        else
-            error("Unrecognized option: $name")
+    fobj = MOI.get(src, MOI.ObjectiveFunction{fobj_type}())
+    obj_coeffs = zeros(length(x_src))
+    obj_offset = 0.0
+
+    if fobj_type == MOI.SingleVariable
+        x = mapping.varmap[fobj.variable]
```

Comment:
You should not support it as Clp does not exploit its structure, it's just treating it as an affine objective